### PR TITLE
Automate release

### PR DESCRIPTION
**Pull request by priyanshuone6**
Merge `update_vcpkg_baseline.py` within `update_version.py`

Add a github workflow to 
- Run `scripts/update_version.py` to
  - Increment version number in
    - `pybamm/version`
    - `docs/conf.py`
    - `CITATION.cff`
    - `vcpkg.json`

  - Update baseline of registries in `vcpkg-configuration.json` as the latest commit id from [pybamm-team/sundials-vcpkg-registry](https://github.com/pybamm-team/sundials-vcpkg-registry)

  - Update `CHANGELOG.md` with a summary of the release
  - Make a PR from develop to main
 - Add a github workflow to automate GitHub releases

### CODE REVIEW

**Comment by priyanshuone6** [comment]
This workflow will create a PR to update versions (see https://github.com/priyanshuone6/PyBaMM/pull/12) and a PR from develop to main (see https://github.com/priyanshuone6/PyBaMM/pull/13) on 28th of every month.

**Comment by priyanshuone6** [comment]
Also, the PR from develop to main should be merged before the month ends, or else the workflow on GitHub releases is set to fail to avoid creating a release for the wrong month.

**Comment by tinosulzer** [comment]
This is a great idea, thanks!
We can maybe get rid of the file `pybamm/version` now, since everything is automatic

**Comment by priyanshuone6** [comment]
Removing `pybamm/version` might not be the optimal solution as it is used to read version in `__init__.py` and `setup.py`. If removed, I would have to open each file and edit the version number in both the files

**Comment by priyanshuone6** [comment]
I have updated the `pybamm/version` just the way sympy uses it in their package. It is also documented in the official [python packaging guide](https://packaging.python.org/en/latest/guides/single-sourcing-package-version/).

File: pybamm/__init__.py [hunk 1]
```diff
@@ -12,37 +12,11 @@
...
-        return year, month
-    except Exception as e:
-        raise RuntimeError("Unable to read version number (" + str(e) + ").")
+from pybamm.version import __version__
 
-
-__version_int__ = _load_version_int()
-__version__ = ".".join(["{:02d}".format(x) for x in __version_int__])
 if sys.version_info[0] < 3:
     del x  # Before Python3, list comprehension iterators leaked
```
Comment (priyanshuone6) [inline pybamm/__init__.py:18]: We should remove these as we no longer support Python 2

Comment (tinosulzer) [reply pybamm/__init__.py:18]: Yes

**Review by tinosulzer — APPROVED** [review]
Looks good to me, ready to merge?

**Comment by priyanshuone6** [comment]
Ready to merge now

**Pull request merged**